File: infra/src/main/java/com/beat/infra/persistence/performanceimage/mapper/PerformanceImagePersistenceMapper.java
package com.beat.infra.persistence.performanceimage.mapper;

import org.springframework.stereotype.Component;

import com.beat.domain.performance.model.PerformanceImage;
import com.beat.infra.persistence.performanceimage.entity.PerformanceImageJpaEntity;

@Component
public class PerformanceImagePersistenceMapper {

	public PerformanceImage toDomain(PerformanceImageJpaEntity entity) {
		return PerformanceImage.rehydrate(
			entity.getId(),
			entity.getPerformanceImageUrl()
		);
	}

	public PerformanceImageJpaEntity toEntity(PerformanceImage performanceImage, long performanceId) {
		return PerformanceImageJpaEntity.rehydrate(
			performanceImage.getId(),
			performanceImage.getPerformanceImageUrl(),
			performanceId
		);
	}
}

File: infrastructure/src/main/kotlin/com/beat/infra/persistence/performanceimage/mapper/PerformanceImagePersistenceMapper.kt
package com.beat.infra.persistence.performanceimage.mapper

import com.beat.domain.performance.model.PerformanceImage
import com.beat.infra.persistence.performanceimage.entity.PerformanceImageJpaEntity
import org.springframework.stereotype.Component

@Component
internal class PerformanceImagePersistenceMapper {
    fun toDomain(entity: PerformanceImageJpaEntity): PerformanceImage = PerformanceImage.rehydrate(
        entity.id,
        entity.performanceImageUrl,
    )

    fun toEntity(performanceImage: PerformanceImage, performanceId: Long): PerformanceImageJpaEntity =
        PerformanceImageJpaEntity.rehydrate(
            performanceImage.id,
            performanceImage.performanceImageUrl,
            performanceId,
        )
}
